lib/check_graphite/projection.rb

```diff
@@ -0,0 +1,39 @@
+require 'attime'
+require 'bigdecimal'
+require 'nagios_check'
+require 'linear-regression'
+
+module CheckGraphite
+  module Projection
+    def self.included(base)
+      base.on '--projection FUTURE_TIMEFRAME', :default => '2days' do |timeframe|
+        options.send('processor=', method(:projected_value))
+        options.send('timeframe=', timeframe)
+      end
+    end
+
+    def projected_value(datapoints)
+      ys, xs = datapoints.transpose
+      lr = Regression::Linear.new(xs, ys)
+      future = CheckGraphite.attime(options.timeframe, xs[-1])
+      value = lr.predict(future.to_i)
+      p = Regression::CorrelationCoefficient.new(xs, ys).pearson
+      store_value options.name, value
+      # Unfortunately, nagios_check converts both primary and
+      # secondary values to float. Hence manual assignment instead:
```

Comment:
I disagree. While comments should be used sparingly, If I break or circumvent a contract, I need to explain why. If you try to change this to `store_value 'p-value', p.abs` you will get an exception from the `undefined` test. Creating a test with the title "make sure the contract is circumvented" is just guessing on the reason for a future breakage; there is no way a test can be that surgical.

If a developer disregards an immediately adjacent comment, that is no different from disregarding an immediately adjacent line of code. I don't really see how a review can objectively object to a useful comment when the review could equally have objected to that comment if it was orphaned.